### PR TITLE
Simplify release process

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

Change the way the release/staging/hotfixing process is done.

instead of using 2 branches, this will be based on tagging.

- master build process is unchanged
- staging is a tag on master branch of type staging_NAMEVERSION, which triggers CI to pull the master images and push them as staging images.
- releasing is a tag on master (MUST BE ALREADY STAGED), which will trigger CI to pull the staging images and push them as release images
- a hotfix is a branch started from a release, that will be deployed **directly** on production without going through staging

- Makefile recipes were added, and the release process documentation was updated.

**IMPORTANT** for @Surfict:  the release tag is changed from for example itisfoundation/webserver:latest to itisfoundation/webserver:release-github-latest

attempt to simplify release process

## How to test

- for testing, the only way is by using your respective fork and your master branch.

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [x] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

**Comment by pcrespov** [comment]
It would be nice to review as well how the changelog is built (see discussion in #1726) when we adopt this approach

**Review by pcrespov — APPROVED** [review]
You implemented nicely the approach we discussed the other day.
I really like the fact that we tag directly in github UI
I took the liberty to suggest some examples to clarify the image tagging
Another thing confusing with the doc is that there is ``git tagging`` and ``docker image tagging``. The first is done by the developer to mark what to release and how and the latter is done by the CI :-)

File: ci/deploy/dockerhub-deploy.bash [hunk 1]
```diff
@@ -1,35 +1,55 @@
...
+
+function error_exit {
+    readonly line=$1
+    shift 1
+    echo
+    echo -e "${RED}[ERROR]:$line: ${1:-"Unknown Error"}" 1>&2
+    exit 1
+}
+
+function log_msg {
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply ci/deploy/dockerhub-deploy.bash:21]: done

File: docs/releasing-workflow-instructions.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  ```
+
+2. Adjust the list of changes if needed
+3. Press the **Publish release** button
+4. The CI will be automatically triggered and will deploy the staging release
+
+## Release process
+
+A released version of simcore, that is marked as such on the master branch by leveraging Github release tagging mechanism. The CI is triggered and will pull the marked staging docker images, and tag them as release images.
+**NOTE:** A release version is ALWAYS preceded by a staging version. The CI will fail if it does not find the corresponding staging version.
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply docs/releasing-workflow-instructions.md:49]: I improved the docs a bit, let me know if it's more clear now.

File: docs/releasing-workflow-instructions.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Each docker build marked as released are tagged as:
+- itisfoundation/[image_name]:release-[CINAME]-latest
+- itisfoundation/[image_name]:release-[CINAME]-v[BUILD_VERSION]-[BUILD_DATE]--[BUILD_TIME].GIT_SHA
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply docs/releasing-workflow-instructions.md]: corrected

File: .github/workflows/ci-testing-deploy.yml [hunk 4]
```diff
@@ -2,7 +2,19 @@ name: Github-CI Push/PR
...
+    branches:
+      - "*"
+    tags-ignore:
+      - "*"
+    paths-ignore:
+      - "*.md"
+      - "*.png"
+      - "*.svg"
+      - "docs/**"
+      - ".vscode/**"
```
Comment (GitHK) [inline .github/workflows/ci-testing-deploy.yml:14]: 👍 I like this, was searching for a way to ignore in the past

File: docs/releasing-workflow-instructions.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+4. The CI will be automatically triggered and will deploy the staging release
+
+## Hotfix process
+
+A hotfix is **ALWAYS made from an already released version**. A branch is created from the tagged release version and the fix is implemented. Once it is ready the CI will trigger and build the *hotfix-* labelled docker images. Then, the image shall be tagged leveraging Github release mechanism. The CI will trigger again and pull the docker images (based on git SHA or latest hotfix image) and push a new release version.
+
+Each docker build marked as released are tagged as:
+
+- itisfoundation/[image_name]:release-[CINAME]-latest
+- itisfoundation/[image_name]:release-[CINAME]-v[RELEASE_VERSION]-[BUILD_DATE]--[BUILD_TIME].[GIT_SHA]
```
Comment (GitHK) [inline docs/releasing-workflow-instructions.md:121]: I see no hotfix in there image name here

File: docs/releasing-workflow-instructions.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+- GIT_SHA is the git SHA corresponding to the code used for building
+
+see [docs/img/git-hotfix-workflow.svg](docs/img/git-hotfix-workflow.svg)
+
+### Hotfix example
+
+A bug was found in version 1.2.0 of the simcore stack. The team decides to fix it as it is a fairly simple one. Therefore a hotfix is created and a new release will be created by the CI:
+
+- ``itisfoundation/webserver:release-github-v1.2.1-2020-09-02--19-30.560eq50f3babb6537580c0e03b85b9a8209bbf10``
+- ``itisfoundation/webserver:release-github-latest``
```
Comment (GitHK) [inline docs/releasing-workflow-instructions.md:138]: same as above, maybe I misunderstood

**Review by GitHK — APPROVED** [review]
Looks fine. Just one minor comment for the docs where I feel something is missing.

File: docs/releasing-workflow-instructions.md (hunk 5 above)
Comment (sanderegg) [reply docs/releasing-workflow-instructions.md:121]: yes the release in the end is a release.
the CI creates a hotfix- candidate release that is pushed to dockerhub

File: docs/releasing-workflow-instructions.md (hunk 6 above)
Comment (sanderegg) [reply docs/releasing-workflow-instructions.md:138]: the result of the hotfix is a release

**Review by odeimaiz — APPROVED** [review]

File: docs/releasing-workflow-instructions.md (hunk 6 above)
Comment (sanderegg) [reply docs/releasing-workflow-instructions.md:132]: added some more docs to the docs ;)

**Pull request merged**